src/building/bootstrapping.md

```diff
@@ -57,6 +57,75 @@ first build the new compiler with an older compiler and then use that to
 build the new compiler with itself. For development, you usually only want
 the `stage1` compiler: `x.py build library/std`.
 
+## Where do stages start and end?
+
+A common question is what exactly happens when you run `x.py build` or `x.py test`.
+Does `--stage 1` mean to _build_ the stage 1 artifacts or to _run_ them?
+In fact, it means both!
+
+<!-- TODO: label each of the cells with the name of the directory in `build` it corresponds to -->
+
+![stages](./stages.png)
+
+So, for example, when you run `x.py test [--stage 1]`,
+that means to build the compiler in row 1 and column 0, then run it on the testsuite.
+This corresponds to the `run-stage` diagram.
+However, when you run `x.py build [--stage 1]`, that means to build the compiler in
+row 2 and column 1. This corresponds to the `build-stage` diagram.
+Building any of the items in the diagram also requires first building all
+items with arrows pointing to it.
+
+The diagram just says `rustc` for simplicity, but this also includes all
+programs linked to rustc:
+
+- `rustdoc`
+- `rustfmt`
+- `clippy`
+- `miri`
+- compiler plugins
+
+Similarly, `std` refers to the whole standard library:
+
+- `core`
+- `alloc`
+- `std`
+- `test`
+- `proc_macro`
+
+### What are `run-stage` and `build-stage`?
```

Comment:
I still don't feel like I fully understand what the distinction between these is.